pliers/extractors/text.py

```diff
@@ -198,3 +206,95 @@ def _extract(self, stim):
         return ExtractorResult(np.array(list(data.values())).transpose(),
                                stim, self, features=list(data.keys()),
                                onsets=onsets, durations=durations)
+
+
+class WordEmbeddingExtractor(TextExtractor):
+
+    ''' An extractor that uses a word embedding file to look up embedding
+    vectors for text.
+
+    Args:
+        embedding_file (str): path to a word embedding file
+        binary (bool): flag indicating whether embedding file is saved in a
+            binary format
+    '''
+
+    _log_attributes = ('wvModel',)
+
+    def __init__(self, embedding_file, binary=False):
+        if KeyedVectors is None:
+            raise ImportError("gensim is required to create a "
+                              "WordEmbeddingExtractor, but could not be "
+                              "successfully imported. Please make sure it is "
+                              "installed.")
+        self.wvModel = KeyedVectors.load_word2vec_format(embedding_file,
+                                                         binary=binary)
+        super(WordEmbeddingExtractor, self).__init__()
+
+    def _extract(self, stim):
+        num_dims = self.wvModel.vector_size
+        if stim.text in self.wvModel:
+            embedding_vector = self.wvModel[stim.text]
+        else:
+            # UNKs will have zeroed-out vectors
+            embedding_vector = np.zeros(num_dims)
+        features = ['embedding_dim%d' % i for i in range(num_dims)]
+        return ExtractorResult([embedding_vector],
+                               stim,
+                               self,
+                               features=features)
+
+
+class TextVectorizerExtractor(BatchTransformerMixin, TextExtractor):
+
+    ''' Uses a scikit-learn Vectorizer to extract bag-of-words
+    features from text.
+
+    Args:
+        vectorizer (sklearn Vectorizer): a scikit-learn Vectorizer to extract
+            with. Will use the CountVectorizer by default.
+    '''
+
+    _log_attributes = ('vectorizer',)
+    _batch_size = sys.maxsize
+
+    def __init__(self, vectorizer=None):
+        super(TextVectorizerExtractor, self).__init__()
+        if vectorizer:
```

Comment:
We may want to check that `vectorizer` is an instance of an sklearn vectorizer. Alternatively (or in addition), we could let it be a string giving the name of a valid sklearn vectorizer, and initialize it for the user, for convenience.